### ISSUE TITLE
Suggestion: require passing 'this' in a constructor

### ISSUE THREAD

**Issue by rix0rrr** [ren-yamanashi/eslint-cdk-plugin#62]
### Feature Description

This library is a fantastic idea! I was going to suggest we create one ourselves, but I'm happy to see we don't need to do that anymore 😉 .

Here's a suggestion for a rule I was considering myself:

Ensure that inside of a Construct constructor, `this` is passed to child constructs instead of `scope`:

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // ❌ Incorrect, passing 'scope' as initial argument
    new Bucket(scope, 'MyBucket');

    // ✅ Correct, should be passing 'this' instead
    new Bucket(this, 'MyBucket');
  }
}
```

### Motivation

I have seen this mistake made in practice

### Invalid Code Examples

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // ❌ Incorrect, passing 'scope' as initial argument
    new Bucket(scope, 'MyBucket');
  }
}
```

### Valid Code Examples

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // ✅ Correct, should be passing 'this' instead
    new Bucket(this, 'MyBucket');
  }
}
```

### Rule Type

problem (possible errors)

### Implementation Details

- [X] This rule is related to AWS CDK best practices
- [ ] I'm willing to submit a PR for this feature

**Comment by ren-yamanashi** [comment]
@rix0rrr 
Thank you for wonderful idea! 
I will incorporate it as a new rule!
